### ISSUE TITLE
Migration snapshot tool creates duplicated index code for UNIQUE indexes

### ISSUE THREAD

**Issue by V-ed** [cakephp/phinx#1237]
Goal : create a migration from an existing database (MySQL)

* Database where migration is being created from : `mysqlnd 5.0.12-dev - 20150407` using `phpMyAdmin`. `PHP version: 7.1.9`

Indexes looks like this in phpMyAdmin's table view :
![indexes of table](https://i.imgur.com/ERnuEPb.png)
Notice how the index `user_id` is `Unique`

When following the [Official CakePHP's migrations cookbook](https://book.cakephp.org/3.0/en/migrations.html#generating-migrations-from-an-existing-database), doing the command `bin/cake bake migration_snapshot Initial` works correctly, but then I had problems [creating an SQLite database](https://book.cakephp.org/3.0/en/migrations.html#migrate-applying-migrations) as the error `General error 1 : user_id_index already exists in [...]` (*it was similar to this, I'm writing this from memory*) and I banged my head on why was that.

I took a look at the code generated by the snapshot command and I saw this :
```php
// [...]
->addIndex(
    [
        'user_id',
    ],
    ['unique' => true]
)
->addIndex(
    [
        'user_id',
    ]
)
// [...]
```

Removing the second `->addIndex()` removes the problem when creating the SQLite database using the migration(s).

### To recap
* Trying to go from MySQL to SQLite using migrations
* Command `bin/cake bake migration_snapshot Initial` generates files without errors
* then trying the command `bin/cake migrations migrate` create the SQLite database with errors :
    * `General error 1 : user_id_index already exists in [...]` in command line
    * only one table exists in SQLite database : `phinxlog`, which has no rows.

I haven't done much testing otherwise.

**Comment by dereuromark** [comment]
PR welcome
